sqlglot/parser.py

```diff
@@ -1194,6 +1192,13 @@ def _parse_property_before(self) -> t.Optional[exp.Expression]:
         return None
 
     def _parse_property(self) -> t.Optional[exp.Expression]:
+        if self._prev.token_type in (
```

Comment:
why is this needed?, why don't you treat it like a normal property

Reply:
Should we factor these tokens out into a class constant set?

Reply:
why don't you just use self._index ?

Reply:
can you describe to me exactly with sql examples why you need this?

Reply:
perfect

Reply:
can you make the change that you proposed and i can review that